kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -125,7 +144,17 @@ class ExecuteStatement(
             setState(OperationState.CANCELED)
 
           case TIMEDOUT_STATE =>
-            setState(OperationState.TIMEOUT)
+            val state =
+              if (operationQueryTimeoutCompatibleState) {
+                if (queryTimeout > 0) {
```

Comment:
do we need this？TIMEDOUT_STATE only happens when `queryTimeout>0` is satisfied on the engine side.
this may be much simpler

case TIMEDOUT_STATE if operationQueryTimeoutCompatibleState => OperationState.CANCELED
case TIMEDOUT_STATE => OperationState.TIMEOUT

Reply:
It was considered before that the server has enabled `queryTimeout`, but although the client supports this feature, `queryTimeout` is not set.
Now change it a little easier, turn it on and convert it.